commands/pull_request.go

```diff
@@ -230,14 +236,23 @@ func pullRequest(cmd *Command, args *Args) {
 Write a message for this pull request. The first block
 of text is the title and the rest is the description.`, fullBase, fullHead))
 
-	if cmd.FlagPassed("message") {
+	switch {
+	case cmd.FlagPassed("message"):
 		messageBuilder.Message = flagPullRequestMessage
 		messageBuilder.Edit = flagPullRequestEdit
-	} else if cmd.FlagPassed("file") {
+	case cmd.FlagPassed("file"):
 		messageBuilder.Message, err = msgFromFile(flagPullRequestFile)
 		utils.Check(err)
 		messageBuilder.Edit = flagPullRequestEdit
-	} else if flagPullRequestIssue == "" {
+	case flagPullRequestNoEdit:
+		commits, _ := git.RefList(baseTracking, head)
+		if len(commits) == 0 {
+			utils.Check(fmt.Errorf("No new commits on branch"))
```

Comment:
Can the wording be more descriptive? E.g. `No commits detected between '<BASE>' and '<HEAD>' branches` (placeholders would be replaced with actual branch names)